insights/parsers/du.py

```diff
@@ -0,0 +1,59 @@
+"""
+du - Command
+============
+
+Module for the processing of output from the ``du`` command stored in
+file.  The base class ``DiskUsage`` provides all of the functionality.
+
+This example demonstrates the funcionality of ``DiskUsage`` class.
+
+Examples:
+    >>> FILE_CONTENT = '''
+    ... COMMAND> du -sh /var/lib/mongodb
+    ...
+    ... 22G     /var/lib/mongodb
+    ... '''
+    >>> du_info = shared[DiskUsage]
+    >>> du_info.get_size('/var/lib/mongodb')
+    22G
+    >>> du_info.get_size('/var/lib/pgsql')
+    None
+
+"""
+
+from .. import Parser, parser, LegacyItemAccess
+from .. import get_active_lines
+
+
+@parser("disk_space")
+class DiskUsage(LegacyItemAccess, Parser):
+    """
+    Typical content of `mongodisk_space` command looks like:
+        COMMAND> du -sh /var/lib/mongodb
+
+        22G     /var/lib/mongodb
+    """
+
+    def __len__(self):
+        return len(self.data)
+
+    def parse_content(self, content):
+        """Parse contents of each line in file which stores ``du`` output.
+
+        Parameters:
+            disk_content : Lines of du output stored in file to be parsed.
+
+        Returns:
+            Dict : A dict of key pair values where key is filesystem path
+            and value is the size of filesystem.
+        """
+
+        data = {}
+        for line in get_active_lines(content, comment_char="COMMAND>"):
+            size, path = line.split()
```

Comment:
This parsing makes a lot of assumptions:
* The path doesn't contain any spaces.
* The line always has at least two parts (what about blank lines mentioned above)
* The line contains a size and a path - what about "/bin/du: Permission denied" or "/bin/du: No such file or directory"?

Reply:
Blank lines are filtered in `get_active_lines`.